### PR TITLE
wallet-ext: use pay and paySui for transferring coins

### PR DESCRIPTION
**Pull request by pchrysochoidis**
* ts-sdk: add transfer functionality to framework coin
* wallet-ext: use pay and paySui for transferring coins

https://user-images.githubusercontent.com/10210143/202043796-51dedd55-6a80-4fc6-acb1-99a69819302d.mov

closes APPS-128

### CODE REVIEW

File: sdk/typescript/src/types/framework.ts [hunk 1]
```diff
@@ -202,6 +203,109 @@ export class Coin {
...
+      coinTypeArg,
+      amountToSend,
+      recipient,
+      gasBudget
+    );
+    return signer.signAndExecuteTransaction(tx);
+  }
+
+  /**
+   * Create a new transaction for sending coins ready to be singed and executed.
```
Comment (666lcz) [inline sdk/typescript/src/types/framework.ts:238]:
```suggestion
   * Create a new transaction for sending coins ready to be signed and executed.
```

File: sdk/typescript/src/types/framework.ts [hunk 2]
```diff
@@ -202,6 +203,109 @@ export class Coin {
...
+   * @param signer User's signer
+   * @param allCoins All the coins that are owned by the user. Can be only the relevant type of coins for the transfer, Sui for gas and the coins with the same type as the type to send.
+   * @param coinTypeArg The coin type argument (Coin<T> the T) of the coin to send
+   * @param amountToSend Total amount to send to recipient
+   * @param recipient Recipient's address
+   * @param gasBudget Gas budget for the tx
+   * @throws in case of insufficient funds, network errors etc
```
Comment (666lcz) [inline sdk/typescript/src/types/framework.ts:245]: I would probably move these comments to the public function above

**Review by 666lcz — APPROVED** [review]

**Comment by 666lcz** [comment]
One edge case we should handle in the wallet is when user wants to transfer all their SUI to another address(and use part of it to pay for the gas). Right now, it's very hard to do so as they need to manually subtract the estimated gas fee from the account balance. We should add a `transfer all` button and use the `PayAllSui` transaction kind

https://github.com/MystenLabs/sui/blob/9bffa79048678f45880455c18c09c993155f9818/crates/sui-types/src/messages.rs#L103-L110

**Comment by pchrysochoidis** [comment]
> One edge case we should handle in the wallet is when user wants to transfer all their SUI to another address(and use part of it to pay for the gas). Right now, it's very hard to do so as they need to manually subtract the estimated gas fee from the account balance. We should add a `transfer all` button and use the `PayAllSui` transaction kind. cc @mystie711 @Jordan-Mysten
> 
> https://github.com/MystenLabs/sui/blob/9bffa79048678f45880455c18c09c993155f9818/crates/sui-types/src/messages.rs#L103-L110

@666lcz  @mystie711 @Jordan-Mysten added [APPS-213](https://mysten.atlassian.net/browse/APPS-213) to track this

**Pull request merged**